messages/dotnet/messages.proto

```diff
@@ -63,14 +63,26 @@ message Meta {
 
   // 386, arm, amd64 etc
   Product cpu = 5;
-  
+
+  // CI environment
+  CI ci = 6;
+
   // A product has a name and a version
   message Product {
     // The product name
     string name = 1;
     // The product version
     string version = 2;
   }
+
+  message CI {
```

Comment:
I'm not sure this is the right name or abstraction. Git-specific information should be in a `Git` message I think. That way we can add support for other source control systems later if needed

I'm not sure what `name` represents, or where it comes from. Same for `projectName`.

How about:

```
message Git {
  string remote = 2;
  string revision = 3;
  string branch = 4;
  string tag = 5;
}
```

The `branch` and `tag` fields should possibly be `repeated` since it's possible to have many.

In practice there will probably only be one, taken from common CI systems' environment variables.

Reply:
That seems a right level of abstraction. I guess notions like the projectName (it would be `cucumber` or `cucumber-ruby` in our case for example) could be extracted from those information.

So something like:

```
message Git {
  string remote = 1;
  string revision = 2;
  string branch = 3;
  string tag = 4;
}

message CI {
  string name = 1;
  string buildId =1;
}
```

I'm not sure about repeating the branch/tags, even if a commit may belong to multiple tags and branch, generally we execute from a given branch or tag.

Reply:
Maybe we should make it clear that this isn't the git info but CI provided git info.

Reply:
Yes that's a good point. I think that would be made clear by nesting the `Git` message inside `CI`.

Reply:
I think I'd prefer a hyperlink instead of a `buildId`. We could name it `url` - it would point to the CI build.

I think we should remove `name` - it can be derived from `remote`